hedera-node/hedera-addressbook-service/src/main/java/com/hedera/node/app/service/addressbook/ReadableNodeStore.java

```diff
@@ -30,6 +31,15 @@
  */
 public interface ReadableNodeStore {
 
+    /**
+     * Constructs a new {@link Roster} object using the current info for each node defined in state.
+     * Accordingly, be warned that <b>this method iterates over all nodes!</b> Only invoke it if
+     * absolutely necessary, e.g. in the case of creating a new roster.
```

Comment:
May be a bit overblown, we only have about 30 nodes, and various constraints suggest we probably won't ever have more than 100, which is not a lot to iterate over.

Reply:
Agreed; it's mostly there to make people think about calling it, but I can soften the language